### ISSUE TITLE
Cannot import <AQGridView.h> into my project even though the files exist

### ISSUE THREAD

**Issue by DaveChambers** [AlanQuatermain/AQGridView#171]
Hi,

I'm trying to use AQGridView in my project and even though I have included the classes AND the demo projects all compile and run I get the error:

'AQGridView.h' file not found.

I followed the instructions for 'How to use it in your project' on github but still no luck.

I'm using Xcode 4.5 and latest SDK.

Can anybody help me out?

Thanks

**Comment by evadne** [comment]
Would you zip the project and put it on-line?
- *

On Oct 7, 2012, at 12:51, DaveChambers notifications@github.com wrote:

Hi,

I'm trying to use AQGridView in my project and even though I have included
the classes AND the demo projects all compile and run I get the error:

'AQGridView.h' file not found.

I followed the instructions for 'How to use it in your project' on github
but still no luck.

I'm using Xcode 4.5 and latest SDK.

Can anybody help me out?

Thanks

—
Reply to this email directly or view it on
GitHubhttps://github.com/AlanQuatermain/AQGridView/issues/171.

**Comment by DaveChambers** [comment]
Hi evadne,

I can't do that as it's a work project and very large also.  Give me 5 minutes and I will make a new project, follow the instructions for 'How to use it in your project' on githuband see if the same error occurs. Then I'll share it to you.

Thanks for replying so fast!

**Comment by DaveChambers** [comment]
Hi evadne,

One thing to note, there is no “Add to Targets” box in Xcode 4.5 as described in the instructions.

Here's a link for another project with same problem:

https://rapidshare.com/files/4077881158/camTest 5.zip

I hope you can help me.

Dave

**Comment by DaveChambers** [comment]
https://rapidshare.com/files/3688114171/camTest_.zip

**Comment by evadne** [comment]
Thanks so much for the sample.  I can take a look in a couple of hours.
 Noticed that you are using angular brackets which are good for framework
headers — try using double quotes instead?
- *

On Oct 7, 2012, at 13:21, DaveChambers notifications@github.com wrote:

https://rapidshare.com/files/3688114171/camTest_.zip

—
Reply to this email directly or view it on
GitHubhttps://github.com/AlanQuatermain/AQGridView/issues/171#issuecomment-9212019.

**Comment by DaveChambers** [comment]
No, I tried both.  I'm sure I have something setup wrong in the Project Settings.  

I'm sure you'll spot it when you take a look.

Thanks

**Comment by evadne** [comment]
Check out [https://github.com/evadne/AQGridView-Issue-171-Sample-1](https://github.com/evadne/AQGridView-Issue-171-Sample-1).  It fixes these things:
-   Uses `$PROJECT_DIR` when describing the header search paths.  It converts to the directory holding the Xcode project.  More info on this StackOverflow question “Canonical list of Xcode Environment Variables” [here](http://stackoverflow.com/questions/6910901/canonical-list-of-xcode-environment-variables).
-   Includes **QuartzCore**.
-   Uses a [Git submodule](http://git-scm.com/book/en/Git-Tools-Submodules) that links to AQGridView instead of a hard copy.

**Comment by DaveChambers** [comment]
Hello,

Thanks for your help.  I found that I could indeed fix camTest by adding AQGridView to my project and simply including QuartzCore - no problems.  I didn't need to follow your first or third bullet point.

However, for my real project I still get the same error.  I additionally edited the User Header Search Paths but it didn't help.

Finally, your last advice:

Uses a Git submodule that links to AQGridView instead of a hard copy.

I didn't fully understand how to do this.  I'd be very grateful if you could provide a step by step guide as to what exactly you did to camTest in that sense.  As I said, it wasn't needed for me with simple projects like camTest but my main project doesn't work so that last step may help.  

I'm really hoping to use AQGridView!  

Thanks, Dave

**Comment by evadne** [comment]
Take a look at Git history in that sample project as that would be the best
I could do as of now.
- *

On Oct 7, 2012, at 21:12, DaveChambers notifications@github.com wrote:

Hello,

Thanks for your help. I found that I could indeed fix camTest by adding
AQGridView to my project and simply including QuartzCore - no problems. I
didn't need to follow your first or third bullet point.

However, for my real project I still get the same error. I additionally
edited the User Header Search Paths but it didn't help.

Finally, your last advice:

Uses a Git submodule that links to AQGridView instead of a hard copy.

I didn't fully understand how to do this. I'd be very grateful if you could
provide a step by step guide as to what exactly you did to camTest in that
sense. As I said, it wasn't needed for me with simple projects like camTest
but my main project doesn't work so that last step may help.

I'm really hoping to use AQGridView!

Thanks, Dave

—
Reply to this email directly or view it on
GitHubhttps://github.com/AlanQuatermain/AQGridView/issues/171#issuecomment-9216046.

**Comment by marmor7** [comment]
I've fixed this by setting the User Header Search paths to: $PROJECT_DIR/AQGridView
and toggling the selector on the right to "recursive".

this makes the search path contain all the sub-dirs within AQGridView, 
which will help it find AQGridView.h in the "Classes" sub-dir.
